### ISSUE TITLE
Cannot find installed generator after fresh install

### ISSUE THREAD

**Issue by howlowck** [yeoman/yo#372]
I got a new computer (El Capitan) recently and installed yeoman.  I've worked with yeoman before and had no issues, but this time, it seems that **yo is not picking up any of the globally installed generator packages.**

I tried installing several generators, including generator-angular, and tried it through npm and the yo CLI installer but to no avail.

Running, `yo doctor` showed all green.  

![image](https://cloud.githubusercontent.com/assets/338265/10713048/afaa8802-7a73-11e5-9fcb-eb11d6d9c5e2.png)

Here is the packages installed globally:
![image](https://cloud.githubusercontent.com/assets/338265/10713055/f6743468-7a73-11e5-8276-46df4564a5f5.png)

But I see no packages under "Run a generator":
![image](https://cloud.githubusercontent.com/assets/338265/10713063/1b47a9b4-7a74-11e5-9188-1944a41dcf19.png)

Is there something that I'm missing? Or is there a way for me to get more details on where an error might have occurred?

**Comment by howlowck** [comment]
also:  
node version: `v4.2.1`  
npm version: `2.14.7`

**Comment by SBoudrias** [comment]
Maybe you have two npm installation conflicting? (although I've never seen that on OSx)

Can you try with the latest npm installed?

**Comment by howlowck** [comment]
@SBoudrias I just have one node/npm installed.

**Comment by eddiemonge** [comment]
Can you try updating your npm?

**Comment by llafuente** [comment]
@howlowck could you please check if `app/index.js` it's in your module.

I lost 2 hours... with this issue.
same node/npm version, downgrade yo doesn't change a thing... 
This should be documented... `package.json` `main` is ignored.

**Comment by SBoudrias** [comment]
> This should be documented... package.json main is ignored.

It is documented plenty http://yeoman.io/authoring/

**Comment by howlowck** [comment]
@eddiemonge thanks for the suggestions, but I've updated my npm and `yo doctor` returns all green.

@llafuente thanks, but I definitely have app/index.js in my own module.  And even when I tried to install `generator-angular` (thru yo's cli or npm -g), yo still doesn't recognize it.

**Comment by llafuente** [comment]
Sorry the offtopic.
@SBoudrias The documentation says:

> The default generator used when you call yo name is the app generator

That's ok, but I don't even use it, so I remove it because it was an empty file.

The documentation should point that in order to yeaoman discover your generator. `app/index.js` must exists, even empty, I know it's mostly a technicality... 

@howlowck 
To throw some ideas.
Check chown/chmod of .npm-packages
Check ~/.npmrc for something not defined in your env
`node -e 'require("generator-xxx");'` shouldn't throw...

**Comment by SBoudrias** [comment]
@llafuente Pretty sure you don't need `app/index.js`. Generators should be discovered without it.

**Comment by howlowck** [comment]
Ok, I found the source of my error.  

I had two yo installed in two different places (probably when I ran `npm install -g` before i configured my .npmrc).  

I deleted the yo that was in the wrong place (CLI was evoking the wrongly placed yo).  Then the yo that was in the same directory as the generator modules was able to pick up the generators.  

Thanks @SBoudrias @llafuente @eddiemonge for helping.
